backend/service/service_block.go

```diff
@@ -135,21 +135,100 @@ func (service *BlockService) QueryBlockInfo(height int64) model.BlockInfo {
 	return result
 }
 
-func (service *BlockService) QueryList(page, size int) model.PageVo {
-	var result []model.BlockInfoVo
-	var pageInfo model.PageVo
+func (service *BlockService) QueryList(page, size int) []model.BlockForList {
+
+	offset := 0
 
-	total, blockList, err := document.Block{}.GetBlockListByPage(page, size)
+	if page > 1 {
+		offset = (page-1)*size - 1
+		size = size + 1
+	}
+
+	blocks, err := document.Block{}.GetBlockListByOffsetAndSize(offset, size)
 
 	if err != nil {
+		logger.Error("GetBlockListByOffsetAndSize", logger.String("err", err.Error()))
 		panic(types.CodeNotFound)
 	}
-	for _, block := range blockList {
-		result = append(result, buildBlock(block))
+
+	addrArr := make([]string, 0, len(blocks))
+
+	for _, v := range blocks {
+		for _, validator := range v.Validators {
+			addrArr = append(addrArr, validator.Address)
+		}
 	}
-	pageInfo.Data = result
-	pageInfo.Count = total
-	return pageInfo
+
+	addrArr = utils.RemoveDuplicationStrArr(addrArr)
+
+	validators, err := document.Validator{}.QueryValidatorMonikerOpAddrConsensusPubkeyByAddr(addrArr)
```

Comment:
parameters of `QueryValidatorMonikerOpAddrConsensusPubkeyByAddr` is `proposalAddrs` not `validatorAddrs`